gabbi/httpclient.py

```diff
@@ -17,12 +17,32 @@
 import os
 import sys
 
-import httplib2
+import urllib3
 
 from gabbi import utils
 
 
-class VerboseHttp(httplib2.Http):
+class Http(urllib3.PoolManager):
+
+    def request(self, absolute_uri, method, body, headers, redirect):
+
+        retries = 1 if redirect else False
```

Comment:
Hm, let me answer your question with a question:

- So how many 302/303s do you want to follow?

> in all cases, if there is any error, don't do a retry and force the "real" error to be raised

So by default, I think urllib3 will try a maximum of 10 (over all things it retries) retries. I think you can disable that elegantly by passing `retries=False` but I've forgotten and haven't used that in a while. I'll need a few minutes to confirm.

If you don't care about following a potentially large number of redirects, you should be able to just say `redirects=True` and be done with all of it.

My example above was allowing for exactly 1 redirect and otherwise returning the status code of the second response (in the event of a redirect) and raising errors otherwise.